BrainPortal/app/models/cluster_task.rb

```diff
@@ -2168,13 +2132,32 @@ def singularity_commands(command_script)
 # by CbrainTask::ClusterTask
 #   #{CbrainTask::ClusterTask.revision_info.to_s}
 
-# CBRAIN internal consistency test
+# CBRAIN internal consistency test 1: must run under proper UID
 if test "$UID" -ne "#{Process.uid}" ; then
   echo "Singularity internal script running with wrong UID (expected UID=#{Process.uid})"
   echo "Runtime IDs: `id`"
   exit 2
 fi
 
+# CBRAIN internal consistency test 2: must have the task workdir mounted inside the container
+if test ! -d #{task_workdir.bash_escape} ; then
+  echo "Container missing mount point for task work directory:" #{task_workdir.bash_escape}
+  exit 2
+fi
+
+# CBRAIN internal consistency test 3: must have the cache_dir mounted inside the container
+if test ! -d #{cache_dir.bash_escape} ; then
+  echo "Container missing mount point for task work directory:" #{cache_dir.bash_escape}
```

Comment:
The message needs to be adjusted, it talks about the task workdir but it's the cache dir instead